filedescriptor/src/windows.rs

```diff
@@ -298,12 +261,76 @@ impl FileDescriptor {
         Ok(())
     }
 
-    pub fn redirect_stdout_to_file(&self) -> anyhow::Result<Redirection> {
-        self.handle.redirect_stdout_to_file()
+    pub(crate) fn redirect_stdin<F: AsRawFileDescriptor>(f: &F) -> anyhow::Result<FileDescriptor> {
+        let raw_std_handle = unsafe { GetStdHandle(STD_INPUT_HANDLE) } as *mut _;
+        let std_original = FileDescriptor {
+            handle: OwnedHandle {
+                handle: raw_std_handle,
+                handle_type: OwnedHandle::probe_handle_type(raw_std_handle),
+            },
+        };
+
+        let cloned_handle = OwnedHandle::dup(f)?;
+        if unsafe {
+            SetStdHandle(
+                STD_INPUT_HANDLE,
+                cloned_handle.handle as *mut winapi::ctypes::c_void,
```

Comment:
I think this should transfer ownership to the OS, otherwise the handle will get closed when `cloned_handle` is dropped when this function returns:

```suggestion
                cloned_handle.into_raw_handle() as *mut winapi::ctypes::c_void,
```

Reply:
Done